orca-core/src/test/groovy/com/netflix/spinnaker/orca/pipeline/util/PackageInfoSpec.groovy

```diff
@@ -16,19 +16,119 @@
+    [fileName: "bar", artifacts: [[:]]]                                                                       || false
+    [fileName: "bat", artifacts: [[:], [:]]]                                                                  || false
+    [fileName: "testFileName", artifacts: [[fileName: "testFileName_1"]]]                                     || true
+    [fileName: "baz", artifacts: [[fileName: "badTestFileName_1"]]]                                           || false
+    [fileName: "testFileName", artifacts: [[fileName: "testFileName_1"], [fileName: "testFileName_2"]]]       || true
+    [fileName: "blah", artifacts: [[fileName: "badTestFileName_1"], [fileName: "badTestFileName_1"]]]         || false
+    [fileName: "testFileName", artifacts: [[fileName: "badTestFileName"], [fileName: "testFileName_1"]]]      || true
+    [fileName: "testFileName", artifacts: [[fileName: "testFileName_1"], [fileName: "badTestFileName"]]]      || true
+  }
+
+  def "If no artifacts in current build info or ancestor stages, code should execute as normal"() {
+
+    given:
+    def quipStage = new PipelineStage(
+      execution: Pipeline.builder().withTrigger(["buildInfo": ["artifacts": [["fileName": "testFileName"]]]]).build(),
+      context: [buildInfo: [name: "someName"], package: "testPackageName"]
+    )
+
+    def packageType = PackageType.DEB
+    def packageInfo =
+      new PackageInfo(quipStage, packageType.packageType, packageType.versionDelimiter, true, false, new ObjectMapper())
+
+    when:
+    packageInfo.findTargetPackage(false)
+
+    then:
+    def ex = thrown(IllegalStateException)
+    ex.message.startsWith("Unable to find deployable artifact starting with")
+  }
+
+  private static Map gc(String name) {
+    [
+      buildInfo: [
+        artifacts: [
+          [fileName: "${name}.0abc298.deb"],
+          [fileName: "${name}.0abc298.war"],
+          [fileName: "build.properties"]
+        ],
+        url: "http://localhost",
+        name: "testBuildName",
+        number: "1"
+      ]
+    ]
+  }
+
+  private static List<StageNavigator.Result> sr(List<String> names) {
+    names.collect() {
+      new StageNavigator.Result(new TestPipelineStage(context: gc(it)))
+    }
+  }
+
+  @Unroll
+  def "If no artifacts in current build info, find from first ancestor that does"() {
+    given:
+    Stage quipStage = new PipelineStage(new Pipeline(), "type", "name", [buildInfo: [name: "someName"], package: "testPackageName"])
+    quipStage.stageNavigator = Mock(StageNavigator) {
+      findAll(_, _) >> { return sr(input) }
+    }
+
+    PackageType packageType = PackageType.DEB
+    PackageInfo packageInfo =
+      new PackageInfo(quipStage, packageType.packageType, packageType.versionDelimiter, true, false, new ObjectMapper())
+
+    when:
+    Map requestMap = packageInfo.findTargetPackage(true)
+
+    then:
+    requestMap.package == "testPackageName_1.0abc298"
```

Comment:
Normally I'd suggest something like ...

```
where:
package   | inputFileNames      || expectedPackage
"appName" | ["appName_1.0.deb"] || "appName_1.0.deb"
...
```

Does that jive with what you're trying to cover in this test?